ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+source "${SHARED_DIR}/govc.sh"
+source "${SHARED_DIR}/vsphere_context.sh"
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+
+readarray -t zones_region_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].region')
+readarray -t zones_zone_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].zone')
+for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+    if [[ $(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.datacenter' | wc -l) == "0" ]];then
+        readarray -t zones_datacenter_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.datacenter')
+    else
+	readarray -t zones_datacenter_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.datacenter')
+    fi
+done
+infra_id=$(oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster)
+
+function check_rhcos() {
+    check_result=0
+    # shellcheck disable=SC2207
+    vsphere_datacenters=($(govc ls /))
+    for vsphere_datacenter in "${vsphere_datacenters[@]}"; do
+        folder_name=$(govc ls ${vsphere_datacenter}/vm | grep ${infra_id})
+	if [ -n "${folder_name}" ]; then
+		echo " folder:${folder_name} under vcenter ${vsphere_datacenter} created successful.  "
+	else
+		check_result=1
+		echo " folder name is incorrect. "
+	fi
+    done	    
+
+    #check all rhcos images uploaded into each failureDomain.
+    match_rhcos=0
+    for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+        rhcos_result=$(govc ls /"${zones_datacenter_from_config[$i]}"/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[${i}]}-${zones_zone_from_config[${i}]} | wc -l)
```

Comment:
same as above comment, need to check if customer folder defined in install-config, then get correct vm folder path.